### ISSUE TITLE
Allow only numbers in dial mode

### ISSUE THREAD

**Issue by jfc-ddev** [dhutaryan/ngx-mat-timepicker#147]
In the latest version it was found:

 - When a character is entered, it no longer detects backspace and the entered value cannot be deleted.

 - It is still allowed to write characters other than numbers.

![image](https://github.com/dhutaryan/ngx-mat-timepicker/assets/156062845/63f64908-206d-4d6c-8b01-f5acea605b2a)

Tip: You could try adding a regex to apply event.preventDefault() when it is verified that it is not a digit.

Greetings

**Comment by dhutaryan** [comment]
Are you sure you updated the version? I checked that button as well. Open the docs and try there. As I see everything works as expected.

**Comment by jfc-ddev** [comment]
Yes, I have version 17.4.1 installed. I'm trying in the dial mode of the documentation, and it lets me enter chars (https://dhutaryan.github.io/ngx-mat-timepicker/#/usage)

![image](https://github.com/dhutaryan/ngx-mat-timepicker/assets/156062845/9fe6a722-c826-469c-b4ec-8670c53e2018)

**Comment by dhutaryan** [comment]
How do you do it? Maybe cache? Because I can't enter anything else except number.

**Comment by jfc-ddev** [comment]
![image](https://github.com/dhutaryan/ngx-mat-timepicker/assets/156062845/a6e97a46-f6e3-423c-b022-4c7a8cdd75df)

I hold down the shift key + one of the following keys [ ] } { -

**Comment by dhutaryan** [comment]
I've tried a lot of combinations on mac and windows and couldn't reproduce it. Did you clean browser cache?

**Comment by jfc-ddev** [comment]
Yes, I did it but the bug is persistent

**Comment by dhutaryan** [comment]
I figure it out. This is some specific behaviour for Spanish keyboard. It ignores `preventDefault` for some reason. Even numeric value check doesn't work. Not sure it will be fixed, at least for now.

**Comment by jfc-ddev** [comment]
Did you try using a regular expression to allow only numbers and then using a preventDefault?

**Comment by dhutaryan** [comment]
Yes, I did. It didn't help as I remember.

**Comment by jfc-ddev** [comment]
Hi

Do you have any news?

**Comment by dhutaryan** [comment]
Nope. I believe the most important is that you cannot apply non-numbers. The fact you can enter into the fields some tricky value that will be replaced on blur is not a big deal.

**Comment by jfc-ddev** [comment]
Hi

Do you have any news?

**Comment by dhutaryan** [comment]
@jfc-ddev nope. Now the exact value is checking and it doesn't work even in this case. There is a magic with Spanish keyboard. Not sure it will be fixed someday.
Anyway, as I said before, the most important that invalid value will be changed to valid after blur.
